### DISCUSSION TITLE
Questions about how to control servos

### DISCUSSION THREAD

**Discussion by kyoeun-space** [bitcraze/discussions#871] [Q&A]
Hello. I've read most of the information about servo, but I want to ask you a few more questions because I'm on a basic level.
I'm going to connect one motor and four servos to Crazyflie bolt 1.1. Four servos will be controlled to fly in the sky with one motor.

One of the ways I found is to use the singlecopter of Ardupilot. Does Crazyflie support Ardupilot?
ardupilot link: https://ardupilot.org/ardupilot/index.html
singlecopter link: https://ardupilot.org/copter/docs/singlecopter-and-coaxcopter.html

If not supported, is it the right way to modify the file in the crayflie2-stm-bootloader? 

Or should I install ST library? If I need to install the library, which path should I install it?

Attached is a brief illustration for understanding. Thank you.

![image](https://github.com/bitcraze/discussions/assets/122524815/c2a44f9a-4f2c-4eab-9c6e-f231b3f51970)

**Comment by tobbeanton** [comment]
> One of the ways I found is to use the singlecopter of Ardupilot. Does Crazyflie support Ardupilot?

 There might be a port to the Crazyflie but I don't know the state of it. You will have to investigate.


> If not supported, is it the right way to modify the file in the crayflie2-stm-bootloader?

We just wrote a bit about our repositories in the most recent [blog post ](https://www.bitcraze.io/2023/08/an-overview-of-our-repositories/)and I suggest digging in to our documentation.

Developing support for a singlecopter should be doable but will require a very good understanding of the Crazyflie system, just to give you a heads up.

**Reply by kyoeun-space** [reply]
I have read the contents of the repository. Thank you!

I've also researched about ardupilot, and it seems to support crazyflie 2.0. However, it seems that a way to support crazyflie 2.1 is under development. I think I need to study more because I'm not sure.

I'll try harder. Thanks!
